build_tools/benchmarks/common/benchmark_driver.py

```diff
@@ -64,8 +64,20 @@ def run(self) -> None:
       self.config.trace_capture_config.capture_tmp_dir.mkdir(parents=True,
                                                              exist_ok=True)
 
-    cpu_target_arch = self.device_info.get_iree_cpu_arch_name()
-    gpu_target_arch = self.device_info.get_iree_gpu_arch_name()
+    use_legacy_name = self.benchmark_suite.legacy_suite
+
+    cpu_target_arch = self.device_info.get_iree_cpu_arch_name(use_legacy_name)
+    if cpu_target_arch is None:
+      print("WARNING: Detected unsupported CPU architecture in "
+            f'"{self.device_info}", CPU benchmarking is disabled.')
+      cpu_target_arch = "unknown"
```

Comment:
Rather than letting magic string filtering take care of this here, should we have an explicit case for running no cpu benchmarks?

Reply:
I realized that we probably just want to provide a list of the included architectures to the `filter_benchmarks_for_category` below. Refactored to avoid the magic string and simplify the interface.